astropy/nddata/utils.py

```diff
@@ -577,11 +576,12 @@ def __init__(self, data, position, size, wcs=None, mode='trim',
                     if wcs is None:
                         raise ValueError('wcs must be input if any element '
                                          'of size has angular units')
-                    if pixel_scales is None:
-                        pixel_scales = u.Quantity(
-                            proj_plane_pixel_scales(wcs), wcs.wcs.cunit[axis])
+                    # proj_plane_pixel_scale return pixel scales in (lon, lat) order
```

Comment:
The order is the same as the axis order in CDELT or CD/PC matrix. It has nothing to do with lon-lat.

Reply:
I see what you mean. However the `size` is given in (lat, lon) order (Numpy convention...), while the cdelt order is typically in (lon, lat), I guess?

Reply:
> while the cdelt order is typically in (lon, lat), I guess

Not necessarily. That is controlled by `CTYPE`. For your example, `wcs.wcs.ctype` is `['RA---CAR', 'DEC--CAR']`, i.e., lon, lat. But you can flip this: `wcs.wcs.ctype = list(wcs.wcs.ctype)[::-1]; wcs.wcs.set()`. It will not change image axes order but it should change what you get when you compute "world" coordinates (`wcs.wcs_pix2world(x, y, 1)`)

Reply:
`size` is always in `(ny, nx)` order.  In general for a rotated WCS, `ny` would not correspond to lon or lat, but some combination of both (that's why `proj_plane_pixel_scales` is used).  As @mcara says, it depends on how the WCS was defined.